### ISSUE TITLE
Change Request: Configure cache in FlatConfig

### ISSUE THREAD

**Issue by ferferga** [eslint/eslint#19015]
### ESLint version

v9.12.0

### What problem do you want to solve?

I'd like to configure different caches depending on an specific package in my monorepo, while also simplify the scripts section in my ``package.json``, which is already unreadable.

Having the option to change cache through config also makes other programmatic configuration easier.

### What do you think is the correct solution?

A ``cache`` object with the following schema could be added to ``linterOptions``:

```js
{
  location: 'path',
  strategy: 'content'
}
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by mdjermanovic** [comment]
Hi @ferferga, thanks for the issue!

We had a similar feature request for the eslintrc config system: https://github.com/eslint/eslint/issues/13938

The request wasn't accepted for the reasons described in https://github.com/eslint/eslint/issues/13938#issuecomment-749877709, and I think the same reasons apply to the flat config system.

@eslint/eslint-team thoughts?

**Comment by ferferga** [comment]
@mdjermanovic Thank you very much for your input and sorry for not finding the original issue!

The original concern is a really valid point (and definitely one I didn't thought the implications of originally). However, I still think this makes sense in Flat Config:

* 4 years ago, monorepo setups were not as ubiquitous as today, but now they're integrated in all common package managers and common. Back then (although I wasn't as knowledgeable of the JS ecosystem as today), I believe most projects just had a single package.json with all the lint commands in one place. Nowadays, package.json can be scattered across the entire repository, making CLI options more difficult to maintain.

* Flat Configs are much more transparent (in fact, I learned a lot about ESLint, and even made some rules, thanks to it!) and easier to track for some mistakes:

    - Knowing what a preset does is as easy as a `console.log`.
    - There are tools like Config Inspector that give a comprehensive insight in what's happening in the config.

* As stated in my original post, FlatConfig being standard JS allows for additional logic that otherwise requires an extra `lint.js` file to pass the appropriate flags. For instance, I have prepared in my internal shareable config the necessary logic to assign the cache depending on the package name of my monorepo, or disabling cache if more than X errors pop out. With JS you are free to do whatever, when editing the JSON you're more limited.

* Moving the `--files` argument from the CLI to the config file was very much appreciated for people! Why not do the same for other CLI options? (In fact, I would pursuit the same thing for all the CLI options, so users have the choice. I'd even do it with env vars, like rclone does!)

* Nothing prevent package authors to modify it, but bad actors will still be bad actors and do wrong in your system regardless. For the good actors, a warning like "you should not modify linter options in your shareable presets".

* Why this concern is (seemingly) non existent with the "report unused directives" option for linter options? Why that one is good, but other CLI flags (including cache) aren't? I haven't found a plugin yet modifying it as well, I believe plugin authors are either:
    A) Aware that this should be only left for the user
    B) Unknowledgeable of this option

* B) leads me to the following conclusion: The report unused directives option is somewhat unknown for beginners to ESLint, so mostly people care about configuring rules in config. However, it's there for somewhat power users, which is what copying this (or all CLI options, as stated above) also exists because this not being request before means most people is fine with CLI options. Normal users will keep their cache settings in CLI (which always takes precedence over configs) like they've been doing already.

**Comment by nzakas** [comment]
Cache behavior, just like autofixing, is a mode that the ESLint CLI runs in. We can't put such options in config files because config files can't specify session-level options, only file-level options. Further, ESLint isn't always run from the CLI, in which case cache options wouldn't make any sense in the config.
